utils/wasm-builder/src/wasm_project.rs

```diff
@@ -91,9 +91,12 @@ impl WasmProject {
             .to_string_lossy()
             .into();
 
+        let target_dir_name = env::var("CARGO_TARGET_DIR")
+            .map(PathBuf::from)
+            .unwrap_or_else(|_err| "target".into());
         let mut target_dir = out_dir.clone();
         while target_dir.pop() {
-            if target_dir.ends_with("target") {
+            if target_dir.ends_with(&target_dir_name) {
                 break;
             }
         }
```

Comment:
should also work with `--target-dir`: https://github.com/rust-lang/cargo/issues/9661#issuecomment-1722358176
```suggestion
        let mut target_dir = out_dir
            .ancestors()
            .find(|path| path.ends_with(&profile))
            .and_then(|path| path.parent().map(|p| p.to_owned()))
            .expect("Could not find target directory");
```

Reply:
Nice find